app/views.py

```diff
@@ -65,14 +72,14 @@ def join(request: HttpRequest):
                 )
         elif game_type == "join.type.types.tournament1v1":
             is_tournament = True
-            if player_count != 4:
+            if player_count % 2 != 0:
                 return HttpResponseBadRequest(
                     "Player count must be a modulo of 2 for this game mode"
                 )
         elif game_type == "join.type.types.tournament1v1v1v1":
             is_tournament = True
             is_match_four = True
-            if player_count != 4:
+            if not player_count % 4 != 0:
```

Comment:
i think using 'not' and '!=' is a bad idea here:
4 % 4 != 0 -> false
not false -> true
(joining with 4 players sends bad request)